utils/lite-components-testkit/src/main/scala/pl/touk/nussknacker/engine/lite/util/test/LiteTestScenarioRunner.scala

```diff
@@ -43,23 +43,21 @@ case class LiteTestScenarioRunner(components: List[ComponentDefinition], config:
     *  .emptySink("sink", LiteTestScenarioRunner.sinkName, "value" -> "#result")
     *  }}}
     */
-  override def runWithData[I:ClassTag, R](scenario: EspProcess, data: List[I]): List[R] = {
-    runWithDataReturningDetails(scenario, data)._2.map(_.result.asInstanceOf[R])
-  }
+  override def runWithData[I:ClassTag, R](scenario: EspProcess, data: List[I]): RunnerResult[R] =
+    runWithDataReturningDetails(scenario, data)
+    .map(result => {
+      RunResult(result._1.map(_.throwable.getMessage), result._2.map(_.result.asInstanceOf[R]))
+    })
 
-  def runWithDataReturningDetails[T: ClassTag](scenario: EspProcess, data: List[T]): (List[ErrorType], List[interpreterTypes.EndResult[AnyRef]]) = {
+  def runWithDataReturningDetails[T: ClassTag](scenario: EspProcess, data: List[T]): SynchronousResult = {
     val testSource = ComponentDefinition(sourceName, new SimpleSourceFactory(Typed[T]))
     val testSink = ComponentDefinition(sinkName, SimpleSinkFactory)
     val (modelData, runId) = ModelWithTestComponents.prepareModelWithTestComponents(config, testSource :: testSink :: components)
     val inputId = scenario.roots.head.id
 
-    try {
-      SynchronousLiteInterpreter
-        .run(modelData, scenario, ScenarioInputBatch(data.map(d => (SourceId(inputId), d))))
-        .run
-    } finally {
-      TestComponentsHolder.clean(runId)
-    }
+    val result = SynchronousLiteInterpreter.run(modelData, scenario, ScenarioInputBatch(data.map(d => (SourceId(inputId), d))))
```

Comment:
It will be safer to keep clean in finally block. We have other unexpected exceptions.